keysign/send.py

```diff
@@ -181,37 +268,74 @@ def on_activate(self, data=None):
     def on_delete_window(*args):
         reactor.callFromThread(reactor.stop)
 
-    
+    def on_toggle_clicked(self, toggle):
+        log.info("Internet toggled to: %s", toggle.get_active())
+        self.send_app.set_internet_option(toggle.get_active())
 
-    def on_key_activated(self, widget, key):
-        ####
-        # Saving subtitle
-        self.headerbar_subtitle = self.headerbar.get_subtitle()
-        self.headerbar.set_subtitle(_("Sending {}").format(key.fpr))
-        ####
-        # Making button clickable
-        self.headerbutton.set_sensitive(True)
+    def on_send_stack_switch(self, stack, *args):
+        log.debug("Switched Send Stack! %r", args)
+        current = self.send_app.stack.get_visible_child()
+        if current == self.send_app.klw:
+            log.debug("Key list page now visible")
+            self.on_keylist_mapped(self.send_app.klw)
+        elif current == self.send_app.kpw:
+            log.debug("Key present page now visible")
+            self.on_keypresent_mapped(self.send_app.kpw)
+        elif current == self.send_app.rb:
+            log.debug("Result page now visible")
+            self.on_resultbox_mapped(self.send_app.rb)
 
```

Comment:
Here I'd like to see an else branch, too.

Reply:
Done with da371c8